ms2deepscore/models/SiameseModel.py

```diff
@@ -105,34 +106,60 @@ def save(self, filename: Union[str, Path]):
-                        l1_reg: float = 1e-6,
-                        l2_reg: float = 1e-6):
+    def get_base_model(input_dim: int,
+                       dims: Tuple[int, ...] = (600, 500, 500),
+                       embedding_dim: int = 400,
+                       dropout_rate: float = 0.25,
+                       l1_reg: float = 1e-6,
+                       l2_reg: float = 1e-6,
+                       dropout_always_on: bool = False) -> keras.Model:
+        """Create base model for Siamaese network.
+
+        Parameters
+        ----------
+        input_dim : int
+            DESCRIPTION.
+        base_dims
+            Tuple of integers depicting the dimensions of the desired hidden
+            layers of the base model
+        embedding_dim
+            Dimension of the embedding (i.e. the output of the base model)
+        dropout_rate
+            Dropout rate to be used in the base model
+        l1_reg
+            L1 regularization rate. Default is 1e-6.
+        l2_reg
+            L2 regularization rate. Default is 1e-6.
+        dropout_always_on
+            Default is False in which case dropout layers will only be active during
+            model training, but switched off during inference. When set to True,
+            dropout layers will always be on, which is used for ensembling via
+            Monte Carlo dropout.
+        """
         # pylint: disable=too-many-arguments
-        model_input = keras.layers.Input(shape=input_dim, name='base_input')
+        model_input = Input(shape=input_dim, name='base_input')
         for i, dim in enumerate(dims):
             if i == 0:
-                model_layer = keras.layers.Dense(dim, activation='relu', name='dense'+str(i+1),
-                                               kernel_regularizer=keras.regularizers.l1_l2(l1=l1_reg, l2=l2_reg))(
+                model_layer = Dense(dim, activation='relu', name='dense'+str(i+1),
+                                    kernel_regularizer=keras.regularizers.l1_l2(l1=l1_reg, l2=l2_reg))(
                    model_input)
             else:
-                model_layer = keras.layers.Dense(dim, activation='relu',
-                                                 name='dense'+str(i+1))(model_layer)
-            model_layer = keras.layers.BatchNormalization(name='normalization'+str(i+1))(model_layer)
-            model_layer = keras.layers.Dropout(dropout_rate, name='dropout'+str(i+1))(model_layer)
+                model_layer = Dense(dim, activation='relu', name='dense'+str(i+1))(model_layer)
+            model_layer = BatchNormalization(name='normalization'+str(i+1))(model_layer)
+            if dropout_always_on:
+                model_layer = Dropout(dropout_rate, name='dropout'+str(i+1))(model_layer,
+                                                                             training=True)
+            else:
+                model_layer = Dropout(dropout_rate, name='dropout'+str(i+1))(model_layer)
```

Comment:
Could be simplified to:
```suggestion

            model_layer = Dropout(dropout_rate, name='dropout'+str(i+1))(model_layer,
                                                                             training=dropout_always_on)
```

Reply:
That could cause trouble, because it would freeze the model in `training=False` for the default state. In that case dropout would stay off even during actual training.